DataConverter/Catalyst.DataConverter/HierarchicalDataTransformer.cs

```diff
@@ -66,38 +79,361 @@ public HierarchicalDataTransformer(IHierarchicalDataTransformerHelper helper)
+            {
+                parentRelationships.AddRange(
+                    otherBinding.ObjectRelationships.Where(
+                        relationship => relationship.ChildObjectId == binding.Id && relationship.ChildObjectType == "Binding").Select(
+                        x => new BindingReference
+                        {
+                            ChildObjectId = x.ChildObjectId,
+                            AttributeValues = x.AttributeValues,
+                            ChildObjectType = x.ChildObjectType,
+                            ParentObjectId = otherBinding.Id
+                        }));
+            }
+
+            LoggingHelper2.Debug(this.guid, $"Found the following parentRelationships for binding with id = {binding.Id}: \n{JsonConvert.SerializeObject(parentRelationships)}");
+
+            return parentRelationships;
+        }
+
+        private async Task<Entity> GetEntityFromBinding(Binding binding)
+        {
+            LoggingHelper2.Debug(this.guid, "Entering GetEntityFromBinding(...)");
+            LoggingHelper2.Debug(this.guid, "binding: " + JsonConvert.SerializeObject(binding));
+
+            if (binding == null || !binding.SourcedByEntities.Any() || binding.SourcedByEntities.FirstOrDefault() == null)
+            {
+                return null;
+            }
+
+            var entityReference = binding.SourcedByEntities.First();
+            var entity = await this.metadataServiceClient.GetEntityAsync(entityReference.SourceEntityId);
+            LoggingHelper2.Debug(this.guid, $"Found source destinationEntity ({entity.EntityName}) for binding (id = {binding.Id})");
+            return entity;
+        }
+
+        private async Task<List<SqlEntityColumnMapping>> GetColumnsFromEntity(Entity sourceEntity, Entity destinationEntity, string entityAlias)
+        {
+            if (sourceEntity == null || destinationEntity == null)
+            {
+                return null;
+            }
+
+            List<SqlEntityColumnMapping> columns = new List<SqlEntityColumnMapping>();
+
+            Field[] sourceEntityFields = await this.metadataServiceClient.GetEntityFieldsAsync(sourceEntity);
+
+            // Add all Active fields (based on destination entity)
+            columns.AddRange(
+                sourceEntityFields
+                    .Where(
+                        field => destinationEntity.Fields.Any(
+                            destinationField => destinationField.FieldName == $"{sourceEntity.EntityName}_{field.FieldName}" && destinationField.Status != FieldStatus.Omitted))
+                    .Select(f => new SqlEntityColumnMapping { Name = f.FieldName, Alias = entityAlias ?? f.FieldName }));
+
+            return columns;
+        }
+
+        private string GetFullyQualifiedTableName(Entity sourceEntity)
+        {
+            return $"[{sourceEntity.DatabaseName}].[{sourceEntity.SchemaName}].[{sourceEntity.TableName}]";
```

Comment:
use the EntityName, not the TableName, as Entityname will point to the view instead of the underlying table